cmake/kokkos_arch.cmake

```diff
@@ -104,12 +104,15 @@ IF(KOKKOS_ENABLE_COMPILER_WARNINGS)
     LIST(APPEND GNU_WARNINGS "-Wimplicit-fallthrough")
   ENDIF()
 
-  COMPILER_SPECIFIC_FLAGS(
-    COMPILER_ID CMAKE_CXX_COMPILER_ID
-    NVHPC       NO-VALUE-SPECIFIED
-    GNU         ${GNU_WARNINGS}
-    DEFAULT     ${COMMON_WARNINGS}
-  )
+  # Not using COMPILER_SPECIFIC_FLAGS function so the warning flags are not passed downstream
+  IF(CMAKE_CXX_COMPILER_ID STREQUAL GNU)
+    STRING(REPLACE ";" " " WARNING_FLAGS "${GNU_WARNINGS}")
+  ELSEIF(CMAKE_CXX_COMPILER_ID STREQUAL NVHPC)
+    # FIXME_NVHPC
+  ELSE()
+    STRING(REPLACE ";" " " WARNING_FLAGS "${COMMON_WARNINGS}")
+  ENDIF()
+  SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${WARNING_FLAGS}")
```

Comment:
We need to make sure no one is using this flag downstream because it can break subdirectory projects. At the very least we should document it.